haskell/src/Ligo/Util.hs

```diff
@@ -0,0 +1,96 @@
+-- SPDX-FileCopyrightText: 2021 Arthur Breitman
+-- SPDX-License-Identifier: LicenseRef-MIT-Arthur-Breitman
+
+-- | Read a contract at compile time.
+module Ligo.Util
+  ( fetchContract
+  , fetchValue
+  , fetchValues
```

Comment:
I'm quite confident we won't need anything beside `fetchContract` from here, so I'd remove the rest.

Reply:
I've removed them as well as making changes according to other suggestion. 

I'm still in the processed of understanding the existing codebase, so I haven't added all the comments/doc yet.

Reply:
@pasqu4le I've added some docs/comments on the type mostly based on LIGO implement and the papers. I could only add some of them since some other fields are not presented in the paper. This should be ready for re-review.